### PR TITLE
Fixed `changeCategoryID()` to not always update the numeric value of the category.

### PR DESCRIPTION
**Pull request by jonkeane**
Added tests of numeric value changes when using `changeCategoryID()` `changeCategoryID()` no longer always makes the numeric value the same as the new ID. This addresses issue #103

### CODE REVIEW

File: R/categories.R [hunk 1]
```diff
@@ -289,9 +291,11 @@ changeCategoryID <- function (variable, from, to) {
 
     ## Add new category
     newcat <- categories(variable)[[pos.from]]
+    if (newcat$id == newcat$numeric_value %||% FALSE) {
+        newcat$numeric_value <- to
```
Comment (GShotwell) [inline R/categories.R]: I wonder if we should have a warning here. I think this is probably the best solution but it's still somewhat counterintuitive behaviour so it might be good to have a warning. It probably is also a good idea to put a comment explaining why this change is here.

Comment (nealrichardson) [inline R/categories.R]: Agree that a code comment is good, but I don't think a user warning is necessary. If users have set their ids and numeric_values to be the same, they'll expect them to remain the same.
